src/clib/initialise_rates.c

```diff
@@ -0,0 +1,457 @@
+
+    //Calculate rate at each temperature.
+    for (int i = 0; i < my_chemistry->NumberOfTemperatureBins; i++){
+        //Set rate to tiny for safety.
+        (*rate_ptr)[i] = tiny;
+
+        //Calculate bin temperature.
+        logT = logT_start + i*d_logT;
+        T = exp(logT);
+
+        //Calculate rate and store.
+        (*rate_ptr)[i] = my_function(T, units, my_chemistry);
+    }
+    return SUCCESS;
+}
+
+//Define a function which will calculate the k13dd rates.
+int add_k13dd_reaction_rate(double **rate_ptr, double units, chemistry_data *my_chemistry)
+{
+    //Allocate array memory to the pointer.
+    *rate_ptr = malloc(14 * my_chemistry->NumberOfTemperatureBins * sizeof(double));
+
+    //Create a temporary array to retrieve the 14 coefficients calculated by the k13dd function.
+    double *temp_array = malloc(14 * sizeof(double));
+
+    //Calculate temperature spacing.
+    double T, logT, logT_start, d_logT;
+    logT_spacing(&logT_start, &d_logT, my_chemistry);
+
+    //Calculate k13dd for all temperatures. Store in results array.
+    for (int i = 0; i < my_chemistry->NumberOfTemperatureBins; i++){
+        //Calculate bin temperature.
+        logT = logT_start + i*d_logT;
+        T = exp(logT);
+
+        //Obtain 14 coefficients at this temperature and place them in results array.
+        k13dd_rate(T, units, temp_array, my_chemistry);
+        for (int j = 0; j < 14; j++){
+            (*rate_ptr)[i + my_chemistry->NumberOfTemperatureBins*(j)] = temp_array[j];
+        }
+    }
+    //Delete temporary array.
+    free(temp_array);
+
+    return SUCCESS;
+}
+    
+//Define a functin which will calculate h2dust rates.
+int add_h2dust_reaction_rate(double **rate_ptr, double units, chemistry_data *my_chemistry)
+{
+    //Allocate memory for h2dust.
+    *rate_ptr = malloc(my_chemistry->NumberOfTemperatureBins * my_chemistry->NumberOfDustTemperatureBins
+                        * sizeof(double));
+
+    //Calculate temperature spacing.
+    double T, logT, logT_start, d_logT, T_dust, logT_dust, logT_start_dust, d_logT_dust;
+    logT_spacing(&logT_start, &d_logT, my_chemistry);
+    logT_spacing_dust(&logT_start_dust, &d_logT_dust, my_chemistry);
+    
+    //Calculate h2dust.
+    for (int i = 0; i < my_chemistry->NumberOfTemperatureBins; i++)
+    {   
+        //Calculate bin temperature.
+        logT = logT_start + i*d_logT;
+        T = exp(logT);
+
+        for (int j = 0; j < my_chemistry->NumberOfDustTemperatureBins; j++)
+        {
+            //Calculate dust bin temperature.
+            logT_dust = logT_start_dust + j*d_logT;
+            T_dust = exp(logT_dust);
+
+            //Calculate rate and store.
+            (*rate_ptr)[i + my_chemistry->NumberOfTemperatureBins*j] = h2dust_rate(T, T_dust, units, my_chemistry);
+        }
+    }
+}
+
+//Definition of the initialise_rates function.
+int initialise_rates(chemistry_data *my_chemistry, chemistry_data_storage *my_rates,
```

Comment:
For the sake of consistency (and totally not because of my Americanism), can you change the spelling to `initialize` to match the other initialize routines?